src/main/java/racingcar/integer_generator/TestIntegerGenerator.java

```diff
@@ -0,0 +1,40 @@
+package racingcar.integer_generator;
+
+import camp.nextstep.edu.missionutils.Randoms;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestIntegerGenerator implements IntegerGenerator {
+
+    private final List<Integer> randomNumberList;
+    private final int carCount;
+    private int currentIndex;
+
+    public TestIntegerGenerator(int carCount) {
+        this.carCount = carCount;
+        this.randomNumberList = createRandomNumberList();
+        this.currentIndex = 0;
+    }
+
+    public List<Integer> getRandomNumberList() {
+        return randomNumberList;
+    }
+
+    @Override
+    public Integer generate() {
+        int lastIndex = randomNumberList.size() - 1;
```

Comment:
Java 21 부터 정해진 순서의 원소를 조회할 수 있는 컬렉션을 표현하는 새로운 인터페이스 `SequencedCollection`가 추가되어 `getFirst()`와 `getLast()` 메서드로 요소를 가져오는게 가능하답니다 😁
```suggestion
        int lastIndex = randomNumberList.getLast();
```
Java 21에 추가된 다른 기능도 찾아보며 활용해봐도 좋을 경험이 될 것 같아요

[Java Docs: SequencedCollection#getLast()](https://docs.oracle.com/en/java/javase/23/docs/api/java.base/java/util/SequencedCollection.html#getLast())